zboxcore/encryption/pre_test.go

```diff
@@ -0,0 +1,133 @@
+package encryption
+
+import (
+	"encoding/base64"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.dedis.ch/kyber/v3/group/edwards25519"
+	"testing"
+)
+
+func TestMnemonic(t *testing.T) {
+	mnemonic := "travel twenty hen negative fresh sentence hen flat swift embody increase juice eternal satisfy want vessel matter honey video begin dutch trigger romance assault"
+
+	encscheme := NewEncryptionScheme()
+
+	err := encscheme.Initialize(mnemonic)
+	require.NoError(t, err)
+
+
+	encscheme.InitForEncryption("filetype:audio")
+	pvk, _ := encscheme.GetPrivateKey()
+	expectedPvk := "XsQLPaRBOFS+3KfXq2/uyAPE+/qq3VW0OkW0T9q93wQ="
+	require.Equal(t, expectedPvk, pvk)
+	pubk, _ := encscheme.GetPublicKey()
+	expectedPubk := "PwpVIXgXbnt8NJmy+R4aSwG8HwJbsbT2JVQqa0bayZQ="
+	require.Equal(t, expectedPubk, pubk)
+
+}
+
+func TestReEncryptionAndDecryptionForShareData(t *testing.T) {
+	client_mnemonic := "travel twenty hen negative fresh sentence hen flat swift embody increase juice eternal satisfy want vessel matter honey video begin dutch trigger romance assault"
+	client_encscheme := NewEncryptionScheme()
+	client_encscheme.Initialize(client_mnemonic)
+	client_encscheme.InitForEncryption("filetype:audio")
+	client_enc_pub_key, err := client_encscheme.GetPublicKey()
+	assert.Nil(t, err)
```

Comment:
Here and elsewhere, use require not assert. We want the test to end on error.

Reply:
@Sriep Okay